src/SignalR/docs/specs/TransportProtocols.md

```diff
@@ -12,19 +12,65 @@ A transport is required to have the following attributes:
 
 ## `POST [endpoint-base]/negotiate` request
 
-The `POST [endpoint-base]/negotiate` request is used to establish a connection between the client and the server. The content type of the response is `application/json`. The response to the `POST [endpoint-base]/negotiate` request contains one of three types of responses:
+The `POST [endpoint-base]/negotiate` request is used to establish a connection between the client and the server.
 
-1. A response that contains the `connectionId` which will be used to identify the connection on the server and the list of the transports supported by the server.
+In the POST request the client sends a query string parameter with the key "negotiateVersion" and the value as the negotiate protocol version it would like to use. If the query string is omitted, the server treats the version as zero. The server will include a "negotiateVersion" property in the json response that says which version it will be using. The version is chosen as described below:
+* If the servers minimum supported protocol version is greater than the version requested by the client it will send an error response and close the connection
+* If the server supports the request version it will respond with the requested version
+* If the requested version is greater than the servers largest supported version the server will respond with its largest supported version
+The client may close the connection if the "negotiateVersion" in the response is not acceptable.
 
+The content type of the response is `application/json` and is a json payload containing properties to assist the client in establishing a persistent connection.
+
+### Version 1
+
+When the server and client agree on version 1 the server response will include a "connectionToken" property in addition to the "connectionId" property. The value of the "connectionToken" property will be used in the "id" query string for the HTTP requests described below, this value should be kept secret.
+
+A successful negotiate response will look similar to the following payload:
+  ```json
+  {
+    "connectionToken":"05265228-1e2c-46c5-82a1-6a5bcc3f0143",
+    "connectionId":"807809a5-31bf-470d-9e23-afaee35d8a0d",
+    "negotiateVersion":1,
+    "availableTransports":[
+      {
+        "transport": "WebSockets",
+        "transferFormats": [ "Text", "Binary" ]
+      },
+      {
+        "transport": "ServerSentEvents",
+        "transferFormats": [ "Text" ]
+      },
+      {
+        "transport": "LongPolling",
+        "transferFormats": [ "Text", "Binary" ]
+      }
+    ]
+  }
+  ```
+
+  The payload returned from this endpoint provides the following data:
+
+  * The `connectionToken` which is **required** by the Long Polling and Server-Sent Events transports (in order to correlate sends and receives).
+  * The `connectionId` which is the id by which other clients can refer to it.
+  * The `negotiateVersion` which is the negotiation protocol version being used between the server and client.
```

Comment:
This name is odd to me. It's more like a `protocolVersion` since it's the version of the protocol, not the version of "negotiation".

Reply:
I think most of us preferred `protocolVersion`, but shied away from from using that since we thought what what the old client used and we wanted to be able to easily distinguish clients from request logs.

It turns out the old client used [clientProtocol](https://github.com/SignalR/SignalR/blob/97f4ad4e2c9312d14cbf09e3d9d25ccd2a354652/src/Microsoft.AspNet.SignalR.JS/jquery.signalR.transports.common.js#L280) as the QS parameter, so I'm also in support of `protocolVersion`.

Reply:
The ASP.NET SignalR negotiate response includes `protocolVersion`

https://github.com/SignalR/SignalR/blob/fc475fa161eaf94150c4aaa4ba71830fd976eb6c/src/Microsoft.AspNet.SignalR.Client/NegotiationResponse.cs#L14